src/utils/fetchStream.ts

```diff
@@ -1,17 +1,61 @@
+import { Flag } from '$/features/flag/types'
+import { MiscAction } from '$/features/misc'
+import { RoomAction } from '$/features/room'
 import { RoomId } from '$/features/room/types'
+import { selectWalletProvider } from '$/features/wallet/selectors'
+import getRoomMetadata from '$/utils/getRoomMetadata'
+import { Provider } from '@web3-react/types'
+import { call, put, select } from 'redux-saga/effects'
 import type StreamrClient from 'streamr-client'
 import type { Stream } from 'streamr-client'
 
-export default async function fetchStream(roomId: RoomId, streamrClient: StreamrClient) {
-    try {
-        const stream: Stream = await streamrClient.getStream(roomId)
+export default function fetchStream(roomId: RoomId, streamrClient: StreamrClient) {
+    return call(function* () {
+        try {
+            const stream: Stream = yield streamrClient.getStream(roomId)
 
-        return stream
-    } catch (e: any) {
-        if (!/NOT_FOUND/.test(e?.message || '')) {
-            throw e
+            const {
+                tokenAddress,
+                tokenIds = [],
+                minRequiredBalance = '0',
+            } = getRoomMetadata(stream)
+
+            yield put(
+                RoomAction.cacheTokenGate({
+                    roomId,
+                    tokenGate: tokenAddress
+                        ? {
+                              tokenAddress,
+                              tokenIds,
+                              minRequiredBalance,
+                          }
+                        : null,
+                })
+            )
+
+            const provider: Provider | undefined = yield select(selectWalletProvider)
+
+            if (!provider) {
+                throw new Error('No provider')
+            }
+
+            if (tokenAddress) {
+                yield put(
```

Comment:
Every time we fetch a room that's gated we also trigger the token standard fetching. This way the upcoming search feature doesn't have to do it separately.

If we know the standard already then it'll be a noop (standard is also cached).